guides/plugins/apps/flow-builder/add-custom-flow-triggers-from-app-system.md

```diff
@@ -0,0 +1,248 @@
+The data given will be saved through `StorableFlow`. This can be utilized for actions or email templates.
+
+*Example to define data from the API:*
+
+```json
+    {
+        "customerId": "d20e4d60e35e4afdb795c767eee08fec",
+        "salesChannelId": "55cb094fd1794d489c63975a6b4b5b90",
+        "shopName": "Shopware's Shop",
+        "url": "https://shopware.com" 
+    }
+```
+
+*After that, at actions we can get data thought FlowStorer.*
+
+```php
+    $salesChanelId = $flow->getData(MailAware::SALES_CHANNEL_ID));
+    $customer = $flow->getData(CustomerAware::CUSTOMER_ID));
+```
+
+*Or we can use the data when defining the email template.*
+
+```html
+    <h3>Welcome to {{ shopName }}</h3>
+    <h1>Visit us at: {{ url }} </h1>
+```
+
+Please see the [StorableFlow Concept](../../../../resources/references/adr/admin/flow-builder/2022-07-21-adding-the-StorableFlow-to-implement-DelayAction-in-Flow-Builder.md) for more information.
+
+## Snippet for translation
+
+You can define snippets to translate your custom trigger to show in the trigger tree and flow list. For more information, please see the  [Adding snippets](../../plugins/administration/adding-snippets.md).
+
+Snippet keys should be defined based on your trigger name defined at `<name>` in your `flow.xml`.
+
+| Fixed key            | Description                                                       |
+|:---------------------|:------------------------------------------------------------------|
+| sw-flow-custom-event | All the keys related to the custom trigger will be defined inside |
+| event-tree           | All the keys used to trigger tree will be defined inside          |
+| flow-list            | All the keys used to flow list will be defined inside             |
+
+***Example***
+
+{% code title="Resources/flow.xml" %}
+
+```xml
+<flow-extensions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/Framework/App/Flow/Schema/flow-1.0.xsd">
+    <flow-events>
+        <flow-event>
+            <name>swag.before.open_the_doors</name>
+            ...
+        </flow-event>
+    </flow-events>
+</flow-extensions>
+```
+
+{% endcode %}
+
+{% code title="custom/apps/FlowBuilderTriggerApp/Resources/app/administration/snippet/en-GB.json" %}
+
+```json
+{
+  "sw-flow-custom-event": {
+    "event-tree": {
+      "swag": "Swag",
+      "before": "Before",
+      "openTheDoors": "Open the doors"
+    },
+    "flow-list": {
+      "swag_before_open_the_doors": "Before open the doors"
+    }
+  }
+}
+```
+
+{% endcode %}
+
+## Install the App
+
+The app can now be installed by running the following command:
+
+```bash
+bin/console app:install --activate FlowBuilderTriggerApp
+```
+
+## Further steps
+
+* [Schema definition for flow events (GitHub)](https://github.com/shopware/platform/blob/trunk/src/Core/Framework/App/Flow/Schema/flow-1.0.xsd)
```

Comment:
What si to be done in this step. Please mention.

Reply:
I have removed the "Further steps" section and relocated the link to "Define the flow trigger."